Makefile

```diff
@@ -952,6 +952,11 @@ ASPP_ERROR = \
 runtime/%.o: runtime/%.S
 	$(ASPP) $(OC_ASPPFLAGS) -o $@ $< || $(ASPP_ERROR)
 
+# A mistake in 5.0 means these are added for compatibility, rather than
+# using $(ocamlrund_CPPFLAGS) and $(ocamlruni_CPPFLAGS).
```

Comment:
It is not clear to me what "A mistake in 5.0 means that ...". This could mean:

- if we were to do a mistake in the future, we would observe ...
- we made a mistake and now we need to ...

I also don't understand the following:
- "for compatibility", for compatibility with what?
- "rather than using ... and ...": again, what does this mean? (that users could have expected to use those instead, maybe this is what they would use on trunk? that we should have named the variables this way from the start, but they are not currently used? something else?)

Reply:
The mistake is that `$(OC_DEBUG_CPPFLAGS)` and `$(OC_INSTR_CPPFLAGS)` are still used in the two recipes just below. On trunk, that's been fixed by using the correct variables `$(ocamlrund_CPPFLAGS)` and `$(ocamlruni_CPPFLAGS)`. However, there's a possible consequence to that - `runtime/amd64.S` contains debugging code that then gets included in `amd64.o` which is not in the present release so, for compatibility with 5.0.0, I suggest that we silence the warning only. There are two ways of doing that: either providing a default (as here) or removing the variable from the recipes. The approach here, pedantically, allows `make OC_DEBUG_CPPFLAGS=-DDEBUG` to allow one to manually build the native runtime with debugging in the assembler stubs enabled. Alternate comment welcomed, but in large part I wasn't expecting it ever to be read, given that it's on the maintenance back-port, not in the original PR!

Reply:
Oh, thinking some more, maybe it's clearer to explain this: there are two bugs. The first is that the version of amd64.o in libasmrund.a does not actually compile the `DEBUG` statements in the sources. The second is that `Makefile` uses undefined variables (`OC_DEBUG_CPPFLAGS` and `OC_INSTR_CPPFLAGS`). On trunk, both bugs are fixed - on 5.0 I propose only fixing the _second_ (since there could be unexpected consequences to fixing the first which don't belong in a point release).

Reply:
Suggestion:

> `$(OC_DEBUG_CPPFLAGS)` and `$(OC_INSTR_CPPFLAGS)` are still used in the two recipes just below. This is a mistake, we should use the correct variables `$(ocamlrund_CPPFLAGS)` and `$(ocamlruni_CPPFLAGS)` instead. This has been fixed on trunk, but in the 5.0 maintenance branch here we just define default values for those OC_* variables to avoid an undefined-variable warning.

Reply:
(I added half a sentence at the end to signify the backward-compatibility intention.)

Reply:
Ta! Slightly amended version pushed